controllers/module_reconciler.go

```diff
@@ -257,6 +262,45 @@ func (r *ModuleReconciler) getNodesListBySelector(ctx context.Context, mod *kmmv
 	return nodes, nil
 }
 
+func (r *ModuleReconciler) injectDTKToBuildIfNeeded(ctx context.Context, kernelVersion string, build *kmmv1beta1.Build) error {
+
+	logger := log.FromContext(ctx, "kernel version")
+
+	if dockerfile := build.Dockerfile; strings.Contains(dockerfile, dtkBuildArg) {
```

Comment:
I would always inject the build argument and not call `strings.Contains` here.

Reply:
IMO. A clear code takes priority to a "fast code" until a bottleneck is created.
I think it is wired (and harder to understand when debugging) to check the buildconfig, see a build arg that doesn't exist in the Dockerfile.

I would stick with adding that build arg only if the user is using the `DTK_AUTO` feature.

Reply:
@yevgeny-shnaidman WDYT? should we always add the build arg or only when we find the substring in the Dockerfile?

Reply:
I personally prefer to go Yoni's way: we are not writing RT application, so strings.Contains performance is negligible in this case, but it does makes code clearer and self-explanatory